### ISSUE TITLE
Stepper component 2.0 update breaks BA Experience builder widget

### ISSUE THREAD

**Issue by ddred11** [Esri/calcite-design-system#8461]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Current updated behavior of the stepper widget breaks our UI for the Business Analyst widget in Experience Builder. At the default width of our widget, user can no longer see additional steps or easily click on them to forward the experience. Making this widget wider is a challenge given the canvas space in Experience Builder.
![Screenshot 2023-12-19 at 2 54 07 PM](https://github.com/Esri/calcite-design-system/assets/5332496/7faf6692-27eb-4b73-9f2b-aa04ba70ac4c). 

### Acceptance Criteria

We would like to have the component provided properties that would allow developers to set the width where the responsive design of the widget trigger so that we can return our widget to this experience.
![Screenshot 2023-12-19 at 2 14 27 PM](https://github.com/Esri/calcite-design-system/assets/5332496/677f1289-0682-4d41-8786-df67e73741a4)

### Relevant Info

Breaks existing functionality. Will affect current users and the apps they have created with this widget.

### Which Component

Stepper component

### Priority impact

p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by macandcheese** [comment]
This will also affect anyone trying to use the Stepper in Modal, as the width is often "below" the current breakpoint to swap to this mode. As a designer / dev user, I want to be able to ensure the Stepper doesn't enter this mode.
![Screenshot 2024-01-04 at 12 20 44 PM](https://github.com/Esri/calcite-design-system/assets/4733155/b05a0423-3088-413b-b697-ad088f5a37ec)

**Comment by SkyeSeitz** [comment]
Issue #8174 is to add the component tokens to do this customization. In the meantime, can we perhaps add the css vars for Stepper Item's `minWidth` value at each scale? @geospatialem Is it possible to prioritize for this milestone? 

cc @alisonailea

**Comment by geospatialem** [comment]
> In the meantime, can we perhaps add the css vars for Stepper Item's `minWidth` value at each scale? @geospatialem Is it possible to prioritize for this milestone?

@SkyeSeitz It looks like there are still some follow-up's if this would indeed solve the use case on the `stepper-item` component, such as would users need to know the number of `stepper-item`s rendered and set a `minWidth` based off the number of items?

Instead, wondering if each breakpoint value defined be overridden to another value, or in cases where the user does not want responsive behavior, have the ability to "zero" out the value?

**Comment by SkyeSeitz** [comment]
Thanks @geospatialem for the clarifying question! 

Regardless of `stepper` width or number of `stepper-items`, the consideration for the user is how narrow do they want to allow `stepper-item`s to get before `stepper` switches to the single view. 

Some users may want to set the `minWidth` to as low as 0px to essentially opt out of the single view. Meaning `stepper-item`s will always continue to squeeze and share inline space as the window narrows. 

On the other hand, some users may still want `stepper` to switch to the single view, but just at a lower threshold where `stepper-item`s are allowed to squeeze further than our default provided `minWidth`. I think this is especially effective, if wanting to build for windows that may change in width or if the `stepper` may be viewed on a mobile device. 

Happy to discuss further as well!

**Comment by ftiru** [comment]
@geospatialem Any update on this issue please? We need this fix for the February release of AGOL as our widget is dependent on this feature please.

**Comment by eriklharper** [comment]
@ftiru Is there a reason why the condensed experience doesn't fit the bill for your scenario?  You can still navigate with the left/right arrows to each of the steps.  

![image](https://github.com/Esri/calcite-design-system/assets/821864/8a450b9f-2594-4639-b5cd-c2fca6dd0d2d)

**Comment by ftiru** [comment]
@eriklharper @geospatialem On production right now, users can create apps like this (narrow BA widget with large map) and easily navigate between the 3 steps of the BA widget by clicking on the steppers/headers as follows which has been the default behavior and that our users are familiar with:

![image](https://github.com/Esri/calcite-design-system/assets/16827336/58aeb83f-a915-438f-b5ac-5ac97650e8e9)

With the recent calcite update, we lost this option to click on the headers for navigation between steps by default. In addition, the stepper labels (Search,  Buffer, Infographics) get hidden even when the widget is still wide as follows:

![image](https://github.com/Esri/calcite-design-system/assets/16827336/92be4a85-fbf1-4d60-a523-d59ba7ed2405)

We have to make our BA widget too wide to get that functionality back. For example here. I had to stretch the BA widget as wide as the map to get the default behavior back:

![image](https://github.com/Esri/calcite-design-system/assets/16827336/fcfaf36f-92b3-44d9-866b-1edf50a29db5)

Basically, the condensed experience does not work well with our widget as it happens even when the widget is wide as follows:

![image](https://github.com/Esri/calcite-design-system/assets/16827336/6d994e99-9f4c-49cb-b769-0fe7f50ec66d)

We need to control when the condensed view should happen to our widget as the current calcite default behavior is not working.

**Comment by eriklharper** [comment]
Curious if you set `scale="s"` if that would allow the step headings to appear at the right width?

**Comment by macandcheese** [comment]
We need to support a way to disable or opt-out of this, or at least "fake opting out" by setting css property / breakpoint overrides. The examples above are clearly worse off for the responsive behavior. 

Setting a different scale just to get this to fit here would have other implications across the UI - inconsistent font size, spacing, etc., when positioned amongst other "m" scale elements.

**Comment by macandcheese** [comment]
We need to support a way to disable or opt-out of this, or at least "fake opting out" by setting css property / breakpoint overrides. The examples above are clearly worse off for the responsive behavior.

Setting a different scale just to get this to fit here would have other implications across the UI - inconsistent font size, spacing, etc., when positioned amongst other "m" scale elements, and would likely be just as fragile as "s" would have its own magic width.

**Comment by eriklharper** [comment]
@ftiru  thanks for the detailed screenshots, these are really helpful!

**Comment by geospatialem** [comment]
Verified in `2.5.1` with https://codepen.io/geospatialem/pen/ZEPVaZL:

![verify-stepper](https://github.com/Esri/calcite-design-system/assets/5023024/cb075ba5-7358-44b1-bff4-732b8c8d8025)
